scalr/eval_and_analysis_pipeline.py

```diff
@@ -180,7 +190,43 @@ def perform_downstream_anlaysis(self):
                 self.flow_logger.exception(e)
 
         if downstream_analysis:
-            self.analysis_config['downstream_analysis'] = downstream_analysis
+            self.analysis_config[
+                'full_samples_downstream_analysis'] = downstream_analysis
+
+    def test_samples_downstream_anlaysis(self):
```

Comment:
If the code is exact same for test samples & full data & we are just fetching datapath from config for full & test, then we can make a single common function here and pass argument datapath from config, so we can avoid duplication of code.

Reply:
Done